apps/fishing-map/features/map/map-sources.hooks.ts

```diff
@@ -0,0 +1,151 @@
+import { useEffect, useMemo } from 'react'
+import { useRecoilValue, useSetRecoilState } from 'recoil'
+import { Feature } from '@globalfishingwatch/maplibre-gl'
+import {
+  ExtendedStyle,
+  HeatmapLayerMeta,
+  TEMPORALGRID_SOURCE_LAYER_INTERACTIVE,
+} from '@globalfishingwatch/layer-composer'
+import {
+  MERGED_ACTIVITY_ANIMATED_HEATMAP_GENERATOR_ID,
+  UrlDataviewInstance,
+} from '@globalfishingwatch/dataviews-client'
+import { MiniglobeBounds } from '@globalfishingwatch/ui-components'
+import useMapInstance from 'features/map/map-context.hooks'
+import { useMapStyle } from 'features/map/map-style.hooks'
+import { mapTilesAtom } from 'features/map/map-sources.atom'
+import { getSourceMetadata } from 'features/map/map-sources.utils'
+import { filterByViewport } from 'features/map/map.utils'
+
+type SourcesHookInput = string | string[]
+
+const sourcesToArray = (sourcesIds: SourcesHookInput) =>
+  Array.isArray(sourcesIds) ? sourcesIds : [sourcesIds]
+
+const getSourcesFromMergedGenerator = (style: ExtendedStyle) => {
+  const meta = getSourceMetadata(style, MERGED_ACTIVITY_ANIMATED_HEATMAP_GENERATOR_ID)
+  return meta.timeChunks.activeSourceId
+}
+
+const getGeneratorSourcesIds = (style: ExtendedStyle, sourcesIds: SourcesHookInput) => {
+  const sourcesIdsList = sourcesToArray(sourcesIds)
+  const sources = sourcesIdsList.flatMap((source) => {
+    if (source === MERGED_ACTIVITY_ANIMATED_HEATMAP_GENERATOR_ID) {
+      return getSourcesFromMergedGenerator(style)
+    }
+    return source
+  })
+  return sources
+}
+
+export const useSourceInStyle = (sourcesIds: SourcesHookInput) => {
+  const style = useMapStyle()
+  if (!sourcesIds || !sourcesIds.length) {
+    return false
+  }
+  const sourcesIdsList = getGeneratorSourcesIds(style, sourcesIds)
+  const sourcesLoaded = sourcesIdsList.every((source) => style?.sources?.[source] !== undefined)
+  return sourcesLoaded
+}
+
+export const useMapSourceTilesLoadedAtom = () => {
+  // Used it once in Map.tsx the listeners only once
+  const map = useMapInstance()
+  const setSourceTilesLoaded = useSetRecoilState(mapTilesAtom)
+
+  useEffect(() => {
+    if (!map) return
+
+    const onSourceDataLoading = (e: any) => {
```

Comment:
Is there an easy way to type `e` ?

Reply:
yes, there is an `MapDataEvent` that we need to extend with the new sourceId property. I'll include in the fork